tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/HotChocolate/ASM/HotChocolateSecurity.cs

```diff
@@ -0,0 +1,175 @@
+// <copyright file="HotChocolateSecurity.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Datadog.Trace.AppSec;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.GraphQL.HotChocolate.ASM.AST;
+using Datadog.Trace.DuckTyping;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.GraphQL.HotChocolate.ASM;
+
+internal abstract class HotChocolateSecurity
+{
+    public static void ScanQuery(IQueryRequest request)
+    {
+        if (Tracer.Instance.ActiveScope is null)
+        {
+            return;
+        }
+
+        // Don't run if ASM isn't enabled or a WebSocket
+        if (!GraphQLSecurityCommon.IsEnabled())
+        {
+            return;
+        }
+
+        var doc = request.Query;
+        if (!doc.TryDuckCast<QueryDocument>(out var document))
+        {
+            return;
+        }
+
+        // Get root node of the document and do a depth search of resolvers
+        foreach (var node in document.Document.Definitions)
+        {
+            try
+            {
+                DepthSearchOperationNode(node, request.VariableValues);
+            }
+            catch
+            {
+                // Failed to search for resolvers with arguments
```

Comment:
what failure are we expecting here? I think logging the exception is probably a good thing.